src/preamble.js

```diff
@@ -1164,6 +1164,16 @@ assert(typeof Int32Array !== 'undefined' && typeof Float64Array !== 'undefined'
 #endif
 
 #if IN_TEST_HARNESS
+
+// Test runs in browsers should always be free from uncaught exceptions. If an uncaught exception is thrown, we fail browser test execution in the REPORT_RESULT() macro to output an error value.
+if (typeof window !== 'undefined') {
+  window.addEventListener('error', function(e) {
+    if (e == 'SimulateInfiniteLoop' || e.message.indexOf('SimulateInfiniteLoop') != -1) return;
```

Comment:
We have a place in postamble.js that just checks the string. Is that incorrect, then?

I think the issue is what is thrown. We throw a string, so it should be ok to check for just the string,
````
$ grep -r "SimulateInfiniteLoop" src/*
src/library_browser.js:      throw 'SimulateInfiniteLoop';
src/library_browser.js:    throw 'SimulateInfiniteLoop';
src/library_glut.js:    throw 'SimulateInfiniteLoop';
src/postamble.js:    } else if (e == 'SimulateInfiniteLoop') {
````
I don't see us throwing it in any other way under `tests/`.

Unless the exception is caught and rethrown, or caught and proxied, or something like that?